caom2repo/caom2repo/core.py

```diff
@@ -602,14 +602,24 @@ def multiprocess_observation_id(collection, observationID, plugin, subject,
                 "{} - To fix the problem, please add the **kwargs "
                 "argument to the list of arguments for the update"
                 " method of your plugin.".format(str(e)))
+        else:
+            # other unexpected TypeError
+            raise e
     except Exception as e:
-        failed = observation.observation_id
-        rootLogger.error('FAILED {} - Reason: {}'.format(observation.observation_id, e))
+        if observation is None:
+            rootLogger.error('Failed to get observation using observationID = {} - Reason: {}'.format(observationID, e))
+            raise e
+        else:
+            failed = observation.observation_id
+            rootLogger.error('FAILED {} - Reason: {}'.format(observation.observation_id, e))
         #if halt_on_error:
         #    raise e TODO
     except KeyboardInterrupt as e:
         # user pressed Control-C or Delete
-        rootLogger.error('FAILED {} - Reason: {}'.format(observation.observation_id, e))
+        if observation is None:
+            rootLogger.error('Failed to get observation using observationID = {} - Reason: {}'.format(observationID, e))
+        else:
+            rootLogger.error('FAILED {} - Reason: {}'.format(observation.observation_id, e))
         raise e
 
     visited = observation.observation_id
```

Comment:
If exception is not re-raised when get_authorization fail, the code will end up here and observation will be None. Safer to replace visited = observationID

Reply:
This block of code is only executed when there is a KeyboardInterrupt, no? visited should only be set when there is no exception/error, no?

Reply:
We increment visited when errors occur on post_observation, don't we. Visited is the total and includes errors or skips.